app/views/play/level/modal/CourseVideosModalComponent.vue

```diff
@@ -0,0 +1,193 @@
+<template lang="pug">
+#modal-base-flat
+  <iframe class="video-frame" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
+  #video-close-btn.btn.well.well-sm.well-parchment.close-btn(data-dismiss="modal")
+    span.glyphicon.glyphicon-remove
+  #course-videos-modal
+    #videos-content.modal-content.style-flat
+      .modal-header
+        #modal-close-btn.btn.well.well-sm.well-parchment.close-btn(data-dismiss="modal")
+          span.glyphicon.glyphicon-remove
+        h3.text-center
+          | {{ $t('courses.concept_videos') }}
+      .modal-body.container
+        .row.videos-row(v-if="videoLevels.length > 0")
+          .col-sm-4.course-video.m-l-3(v-for="level in videoLevels")
+            .locked(v-if="level.videoStatus === 'locked'")
+              .video-link
+                img.video-image(
+                  :src='level.thumbnail_locked'
+                )
+                .video-status
+                  img.img-locked(src = '/images/pages/play/modal/locked_banner.png')
+              .video-text.m-t-2
+                span.rtl-allowed.video-title.gray-text
+                  | {{ level.title }}
+                p.video-desc.gray-text
+                  | {{ $t('courses.locked_videos_desc', { concept_name: level.title }) }}
+            .unlocked(v-else-if="level.videoStatus === 'unlocked'")
+              .video-link
+                img.video-image(
+                  :src='level.thumbnail_unlocked', 
+                  @click="onImageClick"
+                )
+                .video-status
+                  img.img-unlocked(src = '/images/pages/play/modal/unlocked_banner.png')
+              .video-text.m-t-2
+                span.rtl-allowed.video-title
+                  | {{ level.title }}
+                p.video-desc
+                  | {{ $t('courses.unlocked_videos_desc', { concept_name: level.title }) }}
+</template>
+
+<script>
+import api from 'core/api';
+import utils from 'core/utils'
+import Player from '@vimeo/player'
+
+export default Vue.extend({
+  props: {
+    courseInstanceID: {
+      type: String,
+      required: true
+    },
+    courseID: {
+      type: String,
+      required: true
+    }
+  },
+  data: () => ({
+    videoLevels: [],
+    vimeoPlayer: {}
+  }),
+  created() {
```

Comment:
Can we use `co` to unwind these nested promises into cleaner yield calls?
https://github.com/codecombat/codecombat/blob/b77b425/app/views/admin/AdminClassroomLevelsComponent.vue#L63

Reply:
For ES6 transpilation, we do not support yields and generators yet. We might need to install another babel library to support and build that code. Will change it if its fairly simple.